src/components/Images.vue

```diff
@@ -186,72 +171,56 @@ export default {
 			this.$el.onmousemove = null
 		},
 		dragHandler(event) {
-			event.preventDefault()
-			const { pageX, pageY } = event
+			const { clientX, clientY } = event
 
-			if (this.dragging && this.zoomRatio > 1 && pageX > 0 && pageY > 0) {
-				const moveX = this.shiftX + (pageX - this.dragX)
-				const moveY = this.shiftY + (pageY - this.dragY)
-				const growX = this.zoomWidth - this.width
-				const growY = this.zoomHeight - this.height
+			if (this.dragging && this.zoomRatio > 1 && clientX > 0 && clientY > 0) {
+				this.posX += (clientX - this.dragX)
+				this.posY += (clientY - this.dragY)
 
-				this.shiftX = Math.min(Math.max(moveX, -growX / 2), growX / 2)
-				this.shiftY = Math.min(Math.max(moveY, -growY / 2), growX / 2)
-				this.dragX = pageX
-				this.dragY = pageY
+				this.dragX = clientX
+				this.dragY = clientY
 			}
 		},
+
+		/**
+		 * Double click handler
+		 */
 		onDblclick() {
 			if (this.zoomRatio > 1) {
 				this.resetZoom()
 			} else {
+				this.animateTransition = true
+				// 0.15 is half of the delta in zoom,
+				// so that multiplied by width/height centers the image
+				this.posX = -(this.$el.width * 0.15)
+				this.posY = -(this.$el.height * 0.15)
 				this.zoomRatio = 1.3
+				// Animation is 100ms so give it double the time to finish
+				setTimeout(() => { this.animateTransition = false }, 200)
 			}
 		},
 	},
 }
 </script>
 
 <style scoped lang="scss">
-$checkered-size: 8px;
-$checkered-color: #efefef;
-
 img {
-	max-width: 100%;
-	max-height: 100%;
 	align-self: center;
 	justify-self: center;
-	// black while loading
-	background-color: #000;
-	// animate zooming/resize
-	transition: height 100ms ease,
-		width 100ms ease,
-		margin-top 100ms ease,
-		margin-left 100ms ease;
-	// show checkered bg on hover if not currently zooming (but ok if zoomed)
-	&:hover {
-		background-image: linear-gradient(45deg, #{$checkered-color} 25%, transparent 25%),
-			linear-gradient(45deg, transparent 75%, #{$checkered-color} 75%),
-			linear-gradient(45deg, transparent 75%, #{$checkered-color} 75%),
-			linear-gradient(45deg, #{$checkered-color} 25%, #fff 25%);
-		background-size: 2 * $checkered-size 2 * $checkered-size;
-		background-position: 0 0, 0 0, -#{$checkered-size} -#{$checkered-size}, $checkered-size $checkered-size;
-	}
-	&.loaded {
-		// white once done loading
-		background-color: #fff;
-	}
-	&.zoomed {
-		position: absolute;
-		max-height: none;
-		max-width: none;
-		z-index: 10010;
-		cursor: move;
-	}
+	position: absolute;
+	width: 100%;
+	height: 100vh;
```

Comment:
Sorry, no. It will enlarge smaller pictures and 100vh is not compatible with all browsers (can be bigger than the device's visible area).
Plus, I'm fairly certain having such hard fixed height will not work for all images ratio :confused:

Reply:
> It will enlarge smaller pictures

That's the idea since users might want to store small previews to save disk space (since the viewer app uses the previews).

> 100vh is not compatible with all browsers (can be bigger than the device's visible area)

Ok, I can change it to `100%` (I've tested and it still works as expected).

> Plus, I'm fairly certain having such hard fixed height will not work for all images ratio

The `object-fit: contain;` is what maintains the aspect ratio.

Reply:
> Ok, I can change it to `100%` (I've tested and it still works as expected).
> The `object-fit: contain;` is what maintains the aspect ratio.

Unfortunately that creates a variety of other issues. Like the lack of transparency.
If you do that, you suddenly have an img that takes the entire area of the screen, so we cannot have the white background anymore. The requirements are:
- Have an image that scale on resize of the screen/window
- Have an image that does _not_ take more than the viewer parent container size
- Have a white background under the image
- Have a checkered background under the image on hover
- Be able to zoom from and to the image using the cursor position and mousewheel
- When zoomed, be able to move the image if it is bigger than the viewer's container

I think that is pretty much it :)

Reply:
> Have an image that scale on resize of the screen/window

@skjnldsv Any idea how to do that while still filling as much space as possible? The way I can think of is conditionally setting either `max-width` and `height`, or `max-height` and `width`.

Reply:
The only way I found is how it was done. max-height and max-width at 100%
There are tons of similar questions on stack-overflow :stuck_out_tongue:

Reply:
@tcitworld this is not resolved.

Reply:
@skjnldsv It should have been fixed in https://github.com/nextcloud/viewer/commit/aa829e43a0ed62d69a0df95799b7ca53165d2c87

Reply:
I'll address in a new comment :)